### DISCUSSION TITLE
Switching the GUI language does not change the ruleset's language?!

### DISCUSSION THREAD

**Discussion by michaelkubina** [kitodo/kitodo-production#6224] [Q&A] [answered]
Not sure, if i understand something wrong...so i just wanted to ask prior to writing an issue:

When i switch the GUI language to e.g. "Englisch", then all GUI elements get translated properly. But when i open the editor the labels we have defined in the ruleset will not change to the new locale. They remain "Deutsch" even though the fallback should have been used, which has not happened. Is this some kind of misconfiguration or is it actually a bug?

As i have understood from the wiki-page and schema: https://github.com/kitodo/kitodo-production/wiki/Regelsatz-3.x / https://github.com/kitodo/kitodo-production/blob/master/Kitodo/rulesets/ruleset.xsd
- if the locale does not match it should fallback to the default `<label>`, meaning the one without a `lang=` attribute?
- if i provide a corresponding `lang=` attribute this should be used instead, right?

While the wiki states, that ISO 639-1 should be used, the example rulesets (e.g. https://github.com/kitodo/kitodo-production/blob/master/Kitodo/rulesets/dfg-viewer.xml) state that ISO 639-2 should be used:

```
            'label' tag:
                Sets a human readable label for an item.

                - optional attribute 'lang':
                        ISO 639 alpha-2 language code - display language to use
                        the label for. If missing, defines the default value
                        used if there is no specific label for one language
```

Since `lang=de` gets properly applied, it seems, that ISO 639-1 is the way to go. Changing it to ISO 639-2(B or T), meaning `lang="deu"` or `lang="ger"` as stated in the example ruleset resulted in the fallback actually being used - so this is consise to the wiki page that only ISO 639-1 is valid. And since its a wrong value the fallback label got applied.

I have experimented and have used `lang="en"` (639-1) or `lang="eng"` (639-2) without any luck. With an "English" GUI i still got german labels. The same applies for the spanisch translation... and `lang="es"`.

So currently the translation mechanic of the ruleset-labels looks somehow broken to me. But then again i am not sure, if there is something else, that should be configured as well?

Anyone any insights?

**Comment by solth** [comment, marked as answer]
Hi @michaelkubina, UI language and metadata language are configured separately. Perhaps that is the cause for the confusion?

**UI language**: 
![Bildschirmfoto 2024-09-09 um 08 47 47](https://github.com/user-attachments/assets/12397cae-58c1-491d-bc3b-d1f3040fca0f)

**Metadata language**:
![Bildschirmfoto 2024-09-09 um 08 48 35](https://github.com/user-attachments/assets/d991bcf9-2e77-4289-937e-27f00c49d1ff)

**Comment by michaelkubina** [comment]
Thank you very much, this has been an oversight on my side. TBH it's kind of an unexpected behaviour, because changing the UI language feels incomplete when the metadata editor does not follow. I would have expected either a coupling of both (if necessesary with an option for decoupling) or a seperate shortcut for it behind the profile dropdown menu. Good to see its not broken, but we should at least adjust the example rulesets then, because the information about ISO 639-2 is incorrect there and ISO 639-1 should be used instead as stated in the wiki.

Thanks again for your quick reply!
